### PR TITLE
Add linter validation for disallowing MSDN references.

### PR DESCRIPTION
**Pull request by aaronroney**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@aaronroney__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/modeler/AutoRest.Swagger/Validation/DisallowMsdnReferences.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class DisallowMsdnReferences : TypedRule<string>
+    {
+        private const string MsdnString = "https://msdn.microsoft.com/";
```
Comment (fearthecowboy) [inline src/modeler/AutoRest.Swagger/Validation/DisallowMsdnReferences.cs]: you might want this a bit wider than that.
- don't only look for https
- `msdn.com` is an alias for `msdn.microsoft.com` (so links might use that)
- there may be more variants than that...

**Comment by brjohnstmsft** [comment]
Whoa, why is this going to be a rule? Until `docs.microsoft.com` is rock solid, there is still going to be reference content on MSDN, is there not?

Also, is there an issue tracking this that would be a better place to have this discussion?

File: src/modeler/AutoRest.Swagger/Validation/DisallowMsdnReferences.cs (hunk 1 above)
Comment (aaronroney) [inline src/modeler/AutoRest.Swagger/Validation/DisallowMsdnReferences.cs]: Good point, thanks!

**Comment by aaronroney** [comment]
While this is a "rule", it is only a warning.  Violating this rule will neither display any errors during generation nor prevent merging.  It will appear as a warning in the VSCode extension, however.

There is currently no issue, but the docs team would like to at least prevent future references to MSDN with the pending move to docs.microsoft.com.

**Comment by tbombach** [comment]
@azuresdkci add to whitelist!

**Comment by aaronroney** [comment]
Abandoning due to merge from upstream.

**Pull request closed without merging**